src/vm/amd64/PInvokeStubs.asm

```diff
@@ -135,4 +143,91 @@ NESTED_ENTRY VarargPInvokeGenILStub, _TEXT
 
 NESTED_END VarargPInvokeGenILStub, _TEXT
 
+;
+; in:
+; InlinedCallFrame (rcx) = pointer to the InlinedCallFrame data, including the GS cookie slot (GS cookie right 
+;                          before actual InlinedCallFrame data)
+;
+;
+LEAF_ENTRY JIT_PInvokeBegin, _TEXT
+        
+        mov             rax, qword ptr [s_gsCookie]
+        mov             qword ptr [rcx], rax
+        add             rcx, SIZEOF_GSCookie
+
+        ;; set first slot to the value of InlinedCallFrame::`vftable' (checked by runtime code)
+        lea             rax,[??_7InlinedCallFrame@@6B@]
+        mov             qword ptr [rcx], rax
+
+        mov             qword ptr [rcx + OFFSETOF__InlinedCallFrame__m_Datum], 0
```

Comment:
Is the rest of the runtime able to deal with NULL `m_Datum` gracefully?

Reply:
Yes. InlinedCallFrame::Init() also intializes it to null (used by the JIT_InitPInvokeFrame helper)

Reply:
But the regular JITed code initializes it non-null before it makes the call...